openedx_webhooks/lib/github/utils.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from openedx_webhooks.lib.github import gh
+
+
+def _get_most_recent_hook(hooks):
+    sorted_hooks = sorted(hooks, lambda h: h.updated_at, reverse=True)
+    return sorted_hooks[0] if sorted_hooks else None
+
+
+def repo_name(repo):
+    return "{0.owner.login}/{0.name}".format(repo)
+
+
+def create_hook(repo, config, events, active):
+    hook = repo.create_hook('web', config, events, active)
+    if not hook:
+        raise Exception(
+            "Can't create {} webhook: {}".format(repo_name, config['url'])
+        )
+    return hook
+
+
+def edit_hook(repo, hook, config, events, active):
+    # TODO: Test
+    result = hook.edit(config, events, active=active)
+    if not result:
+        raise Exception("Can't edit {} webhook({}): {}".format(
+            repo_name(repo), hook.id, hook.config['url']
+        ))
+
+
+def delete_hook(repo, hook):
+    result = hook.delete()
+    if not result:
+        raise Exception("Can't delete {} webhook({}): {}".format(
+            repo_name(repo), hook.id, hook.config['url']
+        ))
+
+
+def delete_hooks(repo, *hooks):
+    for hook in hooks:
+        delete_hook(repo, hook)
+
+
+def get_webhooks(repo, payload_url):
+    # TODO: Test
+    hooks = []
+    for hook in (hook for hook in repo.iter_hooks() if hook.name == 'web'):
+        if hook.config['url'] == payload_url:
+            hooks.append(hook)
+    return hooks
+
+
+def repo_contains_webhook(repo, payload_url, include_not_active=False):
+    # TODO: Test
+    hooks = get_webhooks(repo, payload_url)
+    is_active = any((h.active for h in hooks))
+    if hooks and (is_active or include_not_active):
+        return True
+    return False
+
+
+def get_repos_with_webhook(payload_url, include_not_active):
+    repos = []
+
+    for repo in gh.iter_repos():
+        if repo_contains_webhook(repo, payload_url, include_not_active):
+            repos.append(repo)
+
+    return set(repos)
```

Comment:
Why make this into a set?  The list seems like exactly what you want. There are no duplicates, right?